### PR TITLE
DSD-1799: Add closeOnBlur to FilterBarInline

### PR DESCRIPTION
**Pull request by jackiequach**
Fixes JIRA ticket [DSD-1799](https://newyorkpubliclibrary.atlassian.net/browse/DSD-1799)

## This PR does the following:

- Adds closeOnBlur prop to the render prop function to apply the appropriate prop value to MultiSelect children.
- Cleans up the FilterBarInline stories by unnesting render prop functions.

## How has this been tested?

- Locally in Storybook

## Accessibility concerns or updates

-

### Accessibility Checklist

- [ ] Checked Storybook's "Accessibility" tab for color contrast and other issues.
- [ ] The feature works with keyboard inputs including tabbing back and forward and pressing space, enter, arrow, and esc keys.
- [ ] For hidden text or when `aria-live` is used, a screenreader was used to verify the text is read.
- [ ] For features that involve UI updates and focusing on DOM `ref`s, focus management was reviewed.
- [ ] The feature works when the page is zoomed in to 200% and 400%.

## Open Questions

-

### Checklist:

- [ ] I have updated the Storybook documentation and changelog accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

[DSD-1799]: https://newyorkpubliclibrary.atlassian.net/browse/DSD-1799?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/FilterBarInline/filterBarInlineChangelogData.ts
Comment (bigfishdesign13) [inline src/components/FilterBarInline/filterBarInlineChangelogData.ts:18]: This is written differently than the entry in the main CHANGELOG file.  Was that intentional?

**Review by bigfishdesign13 — APPROVED** [review]
This looks good to me and it works well, but I would like to get second review.

File: src/components/FilterBarInline/FilterBarInline.mdx
Comment (EdwinGuzman) [inline src/components/FilterBarInline/FilterBarInline.mdx:329]: Typo: the `closeOnBlur` should be within the object.

File: src/components/FilterBarInline/FilterBarInline.mdx
Comment (EdwinGuzman) [inline src/components/FilterBarInline/FilterBarInline.mdx:365]: Hmm my comment above about it being a typo now makes sense. I noticed this in some of the stories, is this because it's for storybook? Otherwise, I'm confused why this is needed.

**Review by EdwinGuzman — COMMENTED** [review]
This looks good but just one question about the `bind` call and why it's needed.

File: src/components/FilterBarInline/FilterBarInline.mdx
Comment (jackiequach) [reply src/components/FilterBarInline/FilterBarInline.mdx:365]: Since `closeOnBlur` is not an argument for `renderMultiSelect` or a prop in `MultiSelectGroup`, this is required to pass it to nested `MultiSelect` components.

Comment (EdwinGuzman) [reply src/components/FilterBarInline/FilterBarInline.mdx:365]: Ah gotcha. Maybe another step is to also allow it in `MultiSelectGroup`? Seems a bit of a rabbit hole for this prop value to go down to `MultiSelect`.

**Review by EdwinGuzman — APPROVED** [review]

**Comment by bigfishdesign13** [comment]
Looks good!

**Pull request merged**
